core/DataAccess/LogAggregator.php

```diff
@@ -1084,67 +1100,92 @@ public function queryConversionsByDimension($dimensions = array(), $where = fals
+        // All user supplied parameters are bound (idSite,  date)
+
+        $sql = sprintf(
+        "SELECT
+          ".$idGoal." AS idgoal,
+          ".($linkField == 'idaction_url' ? '1' : '4')." AS `type`,
+          idaction AS idaction,
+          COUNT(*) AS `1`,     
+          ROUND(SUM(yyy.revenue_total),2) AS `2`,
+          COUNT(yyy.idvisit) AS `3`,
+          ROUND(SUM(yyy.revenue_subtotal),2) AS `4`,
+          ROUND(SUM(yyy.revenue_tax),2) AS `5`,
+          ROUND(SUM(yyy.revenue_shipping),2) AS `6`,
+          ROUND(SUM(yyy.revenue_discount),2) AS `7`,
+          SUM(yyy.items) AS `8`,
+          yyy.pages_before AS `9`,
+          SUM(yyy.attribution) AS `10`,
+          COUNT(*) AS `12`,
+          ROUND(SUM(yyy.revenue),2) AS `15`  
+        FROM (
+          SELECT
+            num_total AS pages_before,
+            1 / num_total AS attribution,
+            r.idvisit AS idvisit,
+            lac.idaction AS idaction,
+            lvcon.revenue AS revenue_total,
+            1 / num_total * lvcon.revenue AS revenue,
+            1 / num_total * lvcon.revenue_subtotal AS revenue_subtotal,
+            1 / num_total * lvcon.revenue_tax AS revenue_tax,
+            1 / num_total * lvcon.revenue_shipping AS revenue_shipping,
+            1 / num_total * lvcon.revenue_discount AS revenue_discount,
+            1 / num_total * lvcon.items AS items
+          FROM (
+            SELECT
+              log_conversion.idvisit,
+              COUNT(*) AS num_total
+            FROM %slog_conversion AS log_conversion FORCE INDEX(index_idsite_datetime)
+            RIGHT JOIN %slog_link_visit_action AS log_vpast ON log_conversion.idvisit = log_vpast.idvisit
+            LEFT JOIN %slog_action lac_past ON log_vpast.".$linkField." = lac_past.idaction
+            WHERE log_conversion.server_time >= ?
+              AND log_conversion.server_time <= ?
+              AND log_conversion.idsite IN (?)
+              AND log_conversion.idgoal = ".$idGoal."
+              AND log_vpast.server_time <= log_conversion.server_time
+              AND lac_past.type = ".($linkField == 'idaction_url' ? '1' : '4')."
+            GROUP BY log_conversion.idvisit
+            ORDER BY NULL
+          ) AS r
+          LEFT JOIN %slog_conversion lvcon ON lvcon.idgoal = ".$idGoal." AND lvcon.idvisit = r.idvisit
+          RIGHT JOIN %slog_link_visit_action logv ON logv.idvisit = r.idvisit
+          LEFT JOIN %slog_action lac ON logv.".$linkField." = lac.idaction
+          WHERE logv.server_time >= ?
+            AND logv.server_time <= ?
+            AND logv.idsite IN (?)
+            AND lac.type = ".($linkField == 'idaction_url' ? '1' : '4')."
+            AND logv.server_time <= lvcon.server_time
+          ) AS yyy
+        GROUP BY yyy.idaction
+        ORDER BY `9` DESC;", $tablePrefix, $tablePrefix, $tablePrefix, $tablePrefix, $tablePrefix, $tablePrefix);
```

Comment:
It looks like this custom query does no longer support segments. The `generateQuery` method did that automatically.
Not sure how to solve that best. Doesn't sound like a good solution to copy over the segment query building here 🤔 
Might be good to also add some tests for the segment use case, to ensure that works correctly.